schema_based_on_diagram.sql

```diff
@@ -0,0 +1,61 @@
+/* Database schema to keep the structure of entire database. */
+CREATE DATABASE clinic;
+
+CREATE TABLE patients(
+id INT GENERATED ALWAYS AS IDENTITY,
+name VARCHAR,
+date_of_birth DATE,
+PRIMARY KEY(id),
+);
+
+CREATE TABLE medical_histories(
+id INT GENERATED ALWAYS AS IDENTITY REFERENCES treatments(id),
+status VARCHAR,
+admitted_at TIMESTAMP,
+patient_id int,
+PRIMARY KEY(id),
+CONSTRAINT fk_patient_id
+FOREIGN KEY (patient_id)
+REFERENCES patients(id)
+ON DELETE CASCADE
+);
+
+CREATE TABLE treatments(
+id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY REFERENCES medical_histories(id),
+type VARCHAR,
+name VARCHAR,
+);
+
+ALTER TABLE medical_histories ADD CONSTRAINT fk_id FOREIGN KEY (id) REFERENCES treatments(id);
+
+ALTER TABLE treatments ADD CONSTRAINT fk_id FOREIGN KEY (id) REFERENCES medical_histories(id);
```

Comment:
- [ ] [OPTIONAL] Good job so far, however I think you are missing the Join table for the many to many relation between `medical_histories` and `treatments` tabel, I am making it as optional, as it was not mentioned earlier but please do implment it as it is a core requirement :+1: 
something like   
```
CREATE TABLE medical_histories_has_treatments (
    medical_history_id int refrences medical_histories(id),
    treatment_id int refrences treatments(id),
    );
```